docs/source/format/CDataInterface.rst

```diff
@@ -292,6 +295,8 @@ are available under the Apache License 2.0.
      void* private_data;
    };
 
+   #endif  // ARROW_C_DATA_H
+
```

Comment:
We could add an explanation here, feel free to elaborate on this suggestion
```suggestion
.. note::
   The canonical guard ``ARROW_C_DATA_INTERFACE`` is meant to avoid duplicate
   definitions if two projects copy the C data interface definitions in their own headers,
   and a third-party project includes from these two projects.
   It is therefore important that this guard is kept exactly as-is when these definitions
   are copied.
```